client/Android/FreeRDPCore/jni/android_freerdp.c

```diff
@@ -964,13 +1064,23 @@ JNIEXPORT void JNICALL jni_freerdp_set_microphone_redirection(JNIEnv *env,
 	settings->AudioCapture = enable;
 	if (enable)
 	{
+		int ret;
 		p = malloc(sizeof(char*));
+		if (!p)
+		{
+			settings->AudioCapture = FALSE;
+			return JNI_FALSE;
+		}
 		p[0] = "audin";
```

Comment:
Like above. Suggest to declare+assign p like this: ```char* p[1] = { "audin" };``` instead of malloc/check/free